safe_transaction_service/history/views.py

```diff
@@ -258,12 +267,61 @@ def list(self, request, *args, **kwargs):
             trusted,
         )
 
-        if not page:
+        return page
+
+    def get_cached_tx_identifiers(self, request, *args, **kwargs) -> Optional[Response]:
+        transaction_service = TransactionServiceProvider()
+        safe = self.kwargs["address"]
+        executed, queued, trusted = self.get_parameters()
+        cache_timeout = settings.CACHE_ALL_TXS_VIEW
+        redis = get_redis()
+
+        # Get all relevant elements for a Safe to be cached
+        relevant_elements = transaction_service.get_count_relevant_txs_for_safe(safe)
+        # Trick to get limit and offset
+        list_pagination = ListPagination(request)
+        limit, offset = list_pagination.limit, list_pagination.offset
+        cache_key = f"all-txs:{int(executed)}{int(queued)}{int(trusted)}:{limit}:{offset}:{safe}:{relevant_elements}"
+        lock_key = f"locks:{cache_key}"
+
+        if not cache_timeout:
+            # Cache disabled
+            return self.get_tx_identifiers(request, *args, **kwargs)
+
+        with redis.lock(
+            lock_key,
+            timeout=60,  # This prevents a service restart to leave a lock forever
```

Comment:
If we have more redis lock I would say to include to environment var. Maybe the same than tasks?

Reply:
Lock should have the same request than request timeout, as it's not expected that a request can lock more, but I need to refactor as Gunicorn configuration cannot be read by the app (as it's outside the package)

Reply:
related refactor https://github.com/safe-global/safe-transaction-service/pull/1609